maliput/tests/phase_ring_book_test.rs

```diff
@@ -71,4 +71,19 @@ fn phase_ring_book_test() {
     assert!(phase_ring.is_some());
     let phase_ring = phase_ring.unwrap();
     assert!(!phase_ring.phases().is_empty());
+
+    let phase = phase_ring.get_phase(&"AllGoPhase".to_string());
+    assert!(phase.is_some());
+    let phase = phase.unwrap();
+    let states = phase.discrete_value_rule_states();
+    assert!(!states.is_empty());
```

Comment:
nit: Can you also check the len of states? to add an extra test. We already know that by checking the yaml